src/ts/components/account/ImportJson.tsx

```diff
@@ -0,0 +1,157 @@
+import * as React from 'react'
+import styled from 'styled-components'
+import t from '../../services/i18n'
+import { importAccountFromJson, decodeAddress } from '../../services/keyring-vault-proxy'
+import { RouteComponentProps, withRouter } from 'react-router'
+import { KeyringPair$Json } from '@polkadot/keyring/types'
+import { Message } from 'semantic-ui-react'
+import Dropzone from 'react-dropzone'
+import { isObject, u8aToString, isHex } from '@polkadot/util'
+import { Section, Button } from '../basic-components'
+
+interface ImportJsonProp extends RouteComponentProps {
+  history: any
+}
+interface ImportJsonState {
+  file?: File
+  json: KeyringPair$Json,
+  password?: string,
+  errorMessage?: string
+}
+
+type LoadEvent = {
+  target: {
+    result: ArrayBuffer
+  }
+}
+
+class ImportJson extends React.Component<ImportJsonProp, ImportJsonState> {
+
+  constructor (props) {
+    super(props)
+    this.handleImport = this.handleImport.bind(this)
+    this.handleFileUpload = this.handleFileUpload.bind(this)
+    this.isReady = this.isReady.bind(this)
+    this.shortFileName = this.shortFileName.bind(this)
+  }
+
+  state: ImportJsonState = {
+    json: {
+      address: '',
+      encoded: '',
+      encoding: { content: ['pkcs8', 'sr25519'], type: 'none', version: '0' },
+      meta: {}
+    }
+  }
+
+  handleImport () {
+    importAccountFromJson(this.state.json, this.state.password)
+      .then((json: KeyringPair$Json) => {
+        console.log(json)
+      }).catch(error => {
+        this.setState({ ...this.state, errorMessage: error })
+      })
+  }
+
+  handleFileUpload (acceptedFiles) {
+    const file = acceptedFiles[0]
+    this.setState({ ...this.state, file: file })
+    const reader = new FileReader()
+    // @ts-ignore
+    reader.onload = ({ target: { result } }: LoadEvent) => {
+      const data = new Uint8Array(result)
+      try {
+        const json = JSON.parse(u8aToString(data))
+        decodeAddress(json.address, true).then(decodeAddress => {
+          if (this.isFileValid(decodeAddress, json)) {
+            this.setState({ ...this.state, json: json, errorMessage: '' })
```

Comment:
show success message?

Reply:
This is resetting errorMessage if user replace a wrong file. Probably we can revisit the UI design when we have the screen linked.